lib/ansible/modules/system/vdo.py

```diff
@@ -0,0 +1,762 @@
+        options.append("--vdoBioThreads=" + params['biothreads'])
+
+    if ('cputhreads' in params) and (params['cputhreads'] is not None):
+        options.append("--vdoCpuThreads=" + params['cputhreads'])
+
+    if ('logicalthreads' in params) and (params['logicalthreads'] is not None):
+        options.append("--vdoLogicalThreads=" + params['logicalthreads'])
+
+    if (('physicalthreads' in params) and
+            (params['physicalthreads'] is not None)):
+        options.append("--vdoPhysicalThreads=" + params['physicalthreads'])
+
+    vdocmdoptions = ' '.join(options)
+    return vdocmdoptions
+
+
+def run_module():
+
+    # Debugging note:
+    # Don't use "print" in ansible modules.  To print debug messages,
+    # add it to the "result" dictionary, the contents of which is printed
+    # when executing "ansible-playbook foo.yml -vvvv".
+
+    # Define the available arguments/parameters that a user can pass to
+    # the module.
+    # Defaults for VDO parameters are None, in order to facilitate
+    # the detection of parameters passed from the playbook.
+    # Creation param defaults are determined by the creation section.
+
+    module_args = dict(
+        name=dict(type='str', required=True),
+        state=dict(choices=['absent', 'present'],
+                   required=False, default='present'),
+        activated=dict(choices=['yes', 'no'],
+                       required=False, default=None),
+        running=dict(choices=['yes', 'no'],
+                     required=False, default=None),
+        growphysical=dict(choices=['yes', 'no'],
+                          required=False, default='no'),
+        device=dict(type='str', required=False),
+        logicalsize=dict(type='str', required=False),
+        deduplication=dict(choices=['enabled', 'disabled'],
+                           required=False, default=None),
+        compression=dict(choices=['enabled', 'disabled'],
+                         required=False, default=None),
+        blockmapcachesize=dict(type='str', required=False),
+        readcache=dict(choices=['enabled', 'disabled'],
+                       required=False, default=None),
+        readcachesize=dict(type='str', required=False),
+        emulate512=dict(choices=['enabled', 'disabled'],
+                        required=False, default=None),
+        slabsize=dict(type='str', required=False),
+        writepolicy=dict(choices=['sync', 'async'],
+                         required=False, default=None),
+        indexmem=dict(type='str', required=False),
+        indexmode=dict(choices=['dense', 'sparse'],
+                       required=False, default=None),
+        ackthreads=dict(type='str', required=False),
+        biothreads=dict(type='str', required=False),
+        cputhreads=dict(type='str', required=False),
+        logicalthreads=dict(type='str', required=False),
+        physicalthreads=dict(type='str', required=False)
```

Comment:
you dont need to include most of the defaults here (nor in docs), `default=None`, `type=str` and `required=false` are all redundant